src/lib/operators/table_scan/base_single_column_table_scan_impl.cpp

```diff
@@ -39,24 +39,50 @@ void BaseSingleColumnTableScanImpl::handle_segment(const ReferenceSegment& segme
   const ChunkID chunk_id = context->_chunk_id;
   auto& matches_out = context->_matches_out;
 
+  const auto& pos_list = segment.pos_list();
+
+  if (pos_list->references_single_chunk() && !pos_list->empty()) {
+    // Fast path :)
+
+    const auto chunk = segment.referenced_table()->get_chunk((*pos_list)[0].chunk_id);
+    auto referenced_segment = chunk->get_segment(segment.referenced_column_id());
+
+    auto new_context = std::make_shared<Context>(chunk_id, matches_out, pos_list);
+
+    resolve_data_and_segment_type(*referenced_segment, [&](const auto data_type_t, const auto& resolved_segment) {
+      static_cast<AbstractSegmentVisitor*>(this)->handle_segment(resolved_segment, new_context);
+    });
+
+    return;
+  }
+
+  // Slow path - we are looking at multiple referenced chunks and need to split the pos list first
   auto referenced_chunk_count = segment.referenced_table()->chunk_count();
-  auto chunk_offsets_by_chunk_id = split_pos_list_by_chunk_id(*segment.pos_list(), referenced_chunk_count);
+  auto chunk_offsets_by_chunk_id = split_pos_list_by_chunk_id(pos_list, referenced_chunk_count);
 
   // Visit each referenced segment
   for (auto referenced_chunk_id = ChunkID{0}; referenced_chunk_id < referenced_chunk_count; ++referenced_chunk_id) {
```

Comment:
This duplicates code with the "Fast Path", can we get rid of that.

Reply:
We probably could, but that would mean that the fast and slow path would be more interwoven and we would have more than one check if we're on the fast path or not. While this would get rid of the duplication, I believe it would hurt the readability.